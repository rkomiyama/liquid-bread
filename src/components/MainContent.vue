<template>
  <v-container>
    <v-layout text-xs-center wrap>
      <v-flex mb-4>
        <SearchBar v-model="search"/>
        <SearchResultsContainer :search="search" :headers="headers"/>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script>
import SearchBar from "./SearchBar";
import SearchResultsContainer from "./SearchResultsContainer";

export default {
  name: "MainContent",
  data() {
    return {
      search: ""
    };
  },
  props: {
    headers: {
      type: Array,
      default: () => []
    }
  },
  components: {
    SearchBar,
    SearchResultsContainer
  }
};
</script>

<style lang="scss" scoped>
</style>
