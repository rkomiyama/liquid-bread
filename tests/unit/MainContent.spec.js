import Vue from "vue";
import Vuetify from "vuetify";
import { shallowMount } from "@vue/test-utils";
import { createRenderer } from "vue-server-renderer";
import MainContent from "@/components/MainContent.vue";

Vue.use(Vuetify);

let wrapper;

describe("MainContent.vue", () => {
  beforeEach(() => {
    wrapper = shallowMount(MainContent);
  });

  it("MainContent has same HTML structure", () => {
    const renderer = createRenderer();
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err);
      expect(str).toMatchSnapshot();
    });
  });
});
