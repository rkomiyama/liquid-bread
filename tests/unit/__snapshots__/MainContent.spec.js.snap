// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent.vue MainContent has same HTML structure 1`] = `
<v-container-stub tag="div">
  <v-layout-stub tag="div" text-xs-center="" wrap="">
    <v-flex-stub tag="div" mb-4="">
      <v-flex-stub tag="div" xs12="" sm8="" md6="" lg4="">
        <v-text-field-stub append-icon="search" label="Search results" value=""></v-text-field-stub>
      </v-flex-stub>
      <SearchResultsContainer-stub search=""></SearchResultsContainer-stub>
    </v-flex-stub>
  </v-layout-stub>
</v-container-stub>
`;
